delivery/common/jwt/examples/jwt-hs256/main.js

```diff
@@ -0,0 +1,33 @@
+import { logger } from 'log';
+import { JWTValidator} from './jwt.js';
+
+//advanced options for jwt validator
+const jwtOption = {
+  //check token expiry
+  ignoreExpiration: false,
+  //check token nbf
+  ignoreNotBefore: false
+}
+
+const jwtValidator = new JWTValidator(jwtOption);
+
+export async function onClientRequest (request) {
+  try {
+    // Fetch hmac veification key from Propery Manager
+    const secretKey = request.getVariable('PMUSER_JWT_HMAC_KEY');
+    let jwt = request.getHeader('Authorization');
+    if (jwt){
+      jwt = jwt[0];
+      const jwtJSON = jwtValidator.decode(jwt);
+      const verifed = await jwtValidator.validate(jwt,jwtJSON.header.alg,secretKey);
```

Comment:
@evan-hughes The reason i had split these function is to handle a case where user might want to verify the token with multiple keys where the success is defined as: token is verified by any one key from the list. 
Having a single function would make the decoding and token claim verification tasks redundant. 
Other solution would be to have a  single function and take a list of keys for verification. Thoughts?

Reply:
> The reason i had split these function is to handle a case where user might want to verify the token with multiple keys where the success is defined as: token is verifiable by any one key from the list.

That makes a lot of sense. I suggest you could support both workflows by providing another toplevel function that would do both the decode-and-validate. It would throw an exception on validation failure (just like the decode failures), and return the decoded headers+payload when all the checks succeed. 

IIRC there's a standard for using the `kid`/key-id field to select keys. A subsequent version could do that.